### ISSUE TITLE
Order clause expression not in select list on MySQL 5.7+

### ISSUE THREAD

**Issue by Draiken** [solidusio/solidus#2016]
### Steps to reproduce
Using MySQL 5.7+ with default configuration
Try to order the products by price on the admin

### Expected behavior
It should order the products by price

### Actual behavior
500 error.
Example of the error:
```
Mysql2::Error: Expression #1 of ORDER BY clause is not in SELECT list, references column 'vff_production.spree_prices.amount' which is not in SELECT list; this is incompatible with DISTINCT: 

SELECT DISTINCT `spree_products`.* 
FROM `spree_products` 
LEFT OUTER JOIN `spree_variants` 
ON `spree_variants`.`product_id` = `spree_products`.`id` 
AND `spree_variants`.`is_master` = 1 
AND `spree_variants`.`deleted_at` IS NULL 
LEFT OUTER JOIN `spree_prices` 
ON `spree_prices`.`variant_id` = `spree_variants`.`id` 
AND `spree_prices`.`currency` = 'BRL' 
AND `spree_prices`.`country_iso` IS NULL 
AND `spree_prices`.`deleted_at` IS NULL 
WHERE `spree_products`.`deleted_at` IS NULL AND (1=1) 
ORDER BY `spree_prices`.`amount` ASC 
LIMIT 10 OFFSET 0
```

### System configuration
**Solidus Version**:
2.2.0
**Extensions in use**:
solidus_auth_devise
solidus_prototypes

Related functionality on MySQL 5.7+: https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by
Note that this is the default SQL mode now. This is also what other DBs already do, so I think it makes sense to fix this.

**Comment by jhawthorn** [comment]
Thanks for the detailed report. It really helps :heart:.

I made some progress reproducing this. I run MariaDB locally, which hasn't changed this default, but it can be turned on `SET sql_mode="ONLY_FULL_GROUP_BY";`.

I think this might be an issue in a few places. I was unable to load the sample data under this `sql_mode`

**Comment by jhawthorn** [comment]
I believe #2025 should fix the original reported issue. I also needed the fix in #2024 in order to load sample data or checkout successfully.
